test/testleakautovar.cpp

```diff
@@ -2828,6 +2828,16 @@ class TestLeakAutoVar : public TestFixture {
               "    SomeClass::someMethod(a);\n"
               "}\n", settingsLeakIgnore);
         ASSERT_EQUALS("[test.cpp:4]: (error) Memory leak: a\n", errout.str());
+
+        check("void bar(int* p) {\n"
+              "    if (p)\n"
+              "        free(p);\n"
+              "}\n"
+              "void f() {\n"
+              "    int* p = malloc(4);\n"
+              "    bar(p);\n"
+              "}\n");
+        ASSERT_EQUALS("", errout.str());
```

Comment:
Something off-topic. I wonder if we also should start adding more negative/positive tests in addition to simply testing for the regression.

I was looking into removing some unnecessary settings in the unit tests but it might not be safe to remove some of those as we might just check for a message not appearing. If that result relies on an option/library being specified the tests would still pass but the message would not be absent because of the fix but because the code in question is never executed.

Reply:
Well `TestLeakAutoVar` sets the `checkLibrary` flag, and there are tests which check the presence of that warning.